chain/network/src/peer.rs

```diff
@@ -605,7 +610,13 @@ impl StreamHandler<Vec<u8>> for Peer {
             self.peer_manager_addr.do_send(metadata);
         }
 
-        peer_msg.record(msg.len());
+        self.network_metrics
```

Comment:
I'm ok putting it behind a flag as stated below, but since I think it should be enabled by default, `metric_recorder` is not a good flag, since we don't want to enable `metric_recorder` by default as it consume more resources.

Reply:
What I am reading here confuses me. It sounds like we have no a very descriptive naming `metric_recorder` if we don't want to include all the recorded metrics under it.

Reply:
What happens in practice, is that metric_recorder store too much information, with little aggregation, it have been useful to track down some issue, but we don't really want to put all metrics there, since some of them should be exposed anyway. I can change the name to `extra_metrics`.

Reply:
`extra_metrics` and `slow_metrics` sound good to me